### PR TITLE
Compile/Link with absolute paths for bundled/vendored deps.

### PR DESCRIPTION
**Pull request by vkalintiris**
##### Summary

The modified options are set only by the netdata-installer.sh script. The paths always point to the correspondent `externaldeps/xxx` directory when the options are specified.

##### Test Plan

CI builds

### CODE REVIEW

**Comment by underhood** [comment]
To provide more info about this PR for reviewers: I assume the motive is to allow `VPATH` automake builds and ultimate goal of this is the GO integration (fixing VPATH builds is of course correct thing with or without golang long term goal)?

File: configure.ac [hunk 1]
```diff
@@ -671,9 +666,9 @@ if test "$enable_cloud" != "no" -a "$aclk_ng" != "yes"; then
 
     if test "${with_bundled_lws}" = "yes"; then
         AC_MSG_CHECKING([if libwebsockets static lib is present])
-        if test -f "${bundled_lws_dir}/libwebsockets.a"; then
-            LWS_CFLAGS="-I ${bundled_lws_dir}/include"
-            OPTIONAL_LWS_LIBS="${bundled_lws_dir}/libwebsockets.a"
+        if test -f "externaldeps/libwebsockets/libwebsockets.a"; then
+            LWS_CFLAGS="-I \$(abs_top_srcdir)/externaldeps/libwebsockets/include"
+            OPTIONAL_LWS_LIBS="externaldeps/libwebsockets/libwebsockets.a"
```
Comment (underhood) [inline configure.ac:671]: why `OPTIONAL_LWS_LIBS` has neigher `abs_top_srcdir` or build dir?

**Comment by vkalintiris** [comment]
> To provide more info about this PR for reviewers: I assume the motive is to allow `VPATH` automake builds and ultimate goal of this is the GO integration (fixing VPATH builds is of course correct thing with or without golang long term goal)?

The primary goal is to support [cgo](https://golang.org/cmd/cgo/). A secondary goal is to standardize the way we are building compilation/linking flags, which in turn will make testing other build systems easier (eg. CMake, Bazel and/or Meson).

Fixing `VPATH` (assuming it's already broken) is outside of the scope of the work I plan to do. Although, in theory this change should make this easier in the future for anyone brave enough to give it a try.

File: configure.ac (hunk 1 above)
Comment (vkalintiris) [reply configure.ac:671]: I plan on doing this in a follow-up PR. I wanted to get a thumbs up first, to make sure everyone's fine with the approach I'm taking. Considering that making the internal library deps use an absolute path is a small change, I can update this PR if you think it's easier/better to review at once.

**Comment by vlvkobal** [comment]
If it's easy to integrate `cgo` to our existing build system then, of course, it's wise doing it that way. But if it's comparable with moving to CMake completely, we should get back to #7150.

**Review by vlvkobal — APPROVED** [review]

**Comment by vkalintiris** [comment]
> If it's easy to integrate `cgo` to our existing build system then, of course, it's wise doing it that way. But if it's comparable with moving to CMake completely, we should get back to #7150.

Thankfully, it's nowhere near as complicated as moving to a new build system :smile: The remaining bits are mostly about calling the go toolchain with the appropriate `CFLAGS` and `LDFLAGS`.

**Comment by vlvkobal** [comment]
> Thankfully, it's nowhere near as complicated as moving to a new build system 😄

BTW, we should still support our CMake build configuration. Will that build work with `cgo` as well as Autotools build?

**Comment by vkalintiris** [comment]
> > Thankfully, it's nowhere near as complicated as moving to a new build system smile
> 
> BTW, we should still support our CMake build configuration. Will that build work with `cgo` as well as Autotools build?

Why should we support CMake?

**Comment by vkalintiris** [comment]
@Ferroin ok to merge this?

**Review by underhood — APPROVED** [review]

**Pull request merged**